src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -732,9 +732,6 @@ internal static void UnregisterMemoryLoadChangeNotification(Action notification)
         /// <typeparam name="T">Specifies the type of the array element.</typeparam>
         /// <param name="length">Specifies the length of the array.</param>
         /// <param name="pinned">Specifies whether the allocated array must be pinned.</param>
-        /// <remarks>
-        /// If pinned is set to true, <typeparamref name="T"/> must not be a reference type or a type that contains object references.
-        /// </remarks>
         [MethodImpl(MethodImplOptions.AggressiveInlining)] // forced to ensure no perf drop for small memory buffers (hot path)
         public static unsafe T[] AllocateUninitializedArray<T>(int length, bool pinned = false) // T[] rather than T?[] to match `new T[length]` behavior
         {
```

Comment:
Maybe remove the check in the `!pinned` path too?

Reply:
Let me know if 13fe7a16ae5407a1b8b4d91f93a14e6fa2d1714b is what you meant.

Reply:
No, I've meant to remove the `RuntimeHelpers.IsReferenceOrContainsReferences<T>()` check there.

Reply:
I see what you mean. Check latest commit.